src/st/create_entities_horizontal.h

```diff
@@ -26,17 +27,35 @@ void CreateEntitiesToTheLeft(s16 posX) {
 #else
     if (true) {
 #endif
-        if (g_LayoutObjHorizontal[LAYOUT_OBJ_POS_X] == LAYOUT_OBJ_START ||
-            posX > g_LayoutObjHorizontal[LAYOUT_OBJ_POS_X]) {
+        layoutObjHorizontal = g_LayoutObjHorizontal;
+        if (layoutObjHorizontal[LAYOUT_OBJ_POS_X] == LAYOUT_OBJ_START) {
+            return;
+        }
+
+        if (posX > layoutObjHorizontal[LAYOUT_OBJ_POS_X]) {
             return;
         }
 
+        flag = (layoutObjHorizontal[LAYOUT_OBJ_SLOT] >> 8);
+#ifdef VERSION_PSP
```

Comment:
I am not very happy with this conditional, but I could not find a way to reconcile the match here for PSP and PSX

See here for the PSP code in PSP decomp which is not matching, if anyone has a good idea here I can resolve this same pattern seen in a few of these files
https://decomp.me/scratch/MH94o